src/core_write.cpp

```diff
@@ -236,6 +237,12 @@ void TxToUniv(const CTransaction& tx, const uint256& hashBlock, UniValue& entry,
     entry.pushKV("version", static_cast<int64_t>(static_cast<uint32_t>(tx.nVersion)));
     entry.pushKV("size", (int)::GetSerializeSize(tx, PROTOCOL_VERSION));
     entry.pushKV("vsize", (GetTransactionWeight(tx) + WITNESS_SCALE_FACTOR - 1) / WITNESS_SCALE_FACTOR);
+    // ELEMENTS: add discountvsize
+    if (Params().GetAcceptDiscountCT()) {
+        entry.pushKV("discountvsize", GetDiscountVirtualTransactionSize(tx));
+    } else {
+        entry.pushKV("discountvsize", (GetTransactionWeight(tx) + WITNESS_SCALE_FACTOR - 1) / WITNESS_SCALE_FACTOR);
+    }
```

Comment:
In 8bafd0c9fa448c3790ed2992f02b8e1245e6c56a:

I think it'd be better to just not have the `discountvsize` field at all, if the node does not accept discount CTs. This is easier to machine-detect and makes the field have only one meaning.

Reply:
done in 1ec3239366b2febe82b62aff23d0069897a5ee8c